src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -566,39 +644,366 @@ public void onFileChange(File file) {
+                throw new RuntimeException(errorMessage);
+            }
+
+            // Read all the output on stdout and return it to the caller
+            BufferedReader in = new BufferedReader(new InputStreamReader(p.getInputStream()));
+            String line;
+            StringBuffer allLines = new StringBuffer();
+            while ((line = in.readLine())!= null) {
+                allLines.append(line);
+            }
+            if (allLines.length() > 0) {
+                result = allLines.toString();
+            }
+        } catch (InterruptedException e) {
+            // Container error, throw exception
+            // If a runtime exception occurred in the server task, log and rethrow
+            error("An interruption error occurred while running a docker command: " + e.getMessage(), e);
+            throw new RuntimeException(e.getMessage());
+        } catch (IOException e) {
+            // Container error, throw exception
+            // If a runtime exception occurred in the server task, log and rethrow
+            error("An error occurred while running a docker command: " + e.getMessage(), e);
+            throw new RuntimeException(e.getMessage());
+        }
+        return result;
+    }
+
+    /**
+     * Build a docker run command with all the ports and directories required to run Open Liberty 
+     * inside a container. Also included is the image name and the server run command to override
+     * the CMD attribute of the Open Liberty docker image. 
+     * @return the command string to use to start the container
+     */
+    private String getContainerCommand() {
+        StringBuffer command = new StringBuffer("docker run --rm");
+        if (httpPort != null) {
+            command.append(" -p "+httpPort+":"+httpPort);
+        } else {
+            command.append(" -p 9080:9080");
+        }
+        debug("System.getProperty(9443) ="+System.getProperty("9443") );
+        if (httpsPort != null) {
+            command.append(" -p "+httpsPort+":"+httpsPort);
+        } else {
+            if (System.getProperty("9443") == null) {
+                command.append(" -p 9443:9443");
+            }
+        }
+        if (libertyDebug) {
+            // map debug port
+            command.append(" -p " + libertyDebugPort + ":" + libertyDebugPort);
+            // set environment variables in the container to ensure debug mode does not suspend the server, and to enable a custom debug port to be used
+            command.append(" -e WLP_DEBUG_SUSPEND=n -e WLP_DEBUG_ADDRESS=" + libertyDebugPort);
+        }
+
+        // mount .war.xml potential directories - override /config/apps and /config/dropins
+        command.append(" -v " + serverDirectory + "/apps:/config/apps");
+        command.append(" -v " + serverDirectory + "/dropins:/config/dropins");
+
+        // mount the loose application resources in the container
+        command.append(" -v "+projectDirectory.getAbsolutePath()+":"+DEVMODE_DIR_NAME);
```

Comment:
getCanonicalPath()

Reply:
This creates the command to pass to Docker, and it is not used for path comparison.  So absolute path should work and it just leaves the path resolution to the command line or Docker.